service_crategen/src/commands/generate/codegen/json.rs

```diff
@@ -91,15 +91,14 @@ impl GenerateProtocol for JsonGenerator {
                     request
                 }}
 
-                async fn sign_and_dispatch<E>(
+                async fn sign_and_dispatch(
                     &self,
                     request: SignedRequest,
-                    from_response: fn (BufferedHttpResponse) -> RusotoError<E>,
-                ) -> Result<HttpResponse, RusotoError<E>> {{
+                ) -> Result<HttpResponse, RusotoError<std::convert::Infallible>> {{
                     let mut response = self.client.sign_and_dispatch(request).await?;
                     if !response.status.is_success() {{
-                        let response = response.buffer().await.map_err(RusotoError::HttpDispatch)?;
```

Comment:
Same comment as for codegen/query.rs -- not sure about the rationale for changing from the `HttpDispatch` variant to `Unknown`.